### PR TITLE
feat/5282 - Auto-submit the Setup Component form having returned from the OAuth flow

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #5282 

## Relevant technical choices

- Some aspects of the IB weren't needed due to the changes in #5278 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.js [hunk 1]
```diff
@@ -337,7 +356,10 @@ export default function SetupBanner( { onSubmitSuccess } ) {
 			className="googlesitekit-ga4-setup-banner"
 			title={ title }
 			ctaComponent={
-				<SpinnerButton onClick={ handleSubmitChanges }>
+				<SpinnerButton
+					onClick={ handleSubmitChanges }
+					forceProcessing={ forceProcessing }
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.js:361]: I can understand why you've added this prop. However, now that we are using a prop to specify the spinner state, it's evident that the `SpinnerButton` is more complicated than it needs to be. Bearing in mind we have only introduced it and are using it in this one case so far, I think we should remove the use of `setState` from `SpinnerButton` and always have the spinner state determined via the prop.

One additional point, I would suggest for consistency with the existing [`Spinner`](https://github.com/google/site-kit-wp/blob/f6d2929825f311086508abd110426735c65e00a6/assets/js/components/Spinner.js#L28) component that we name the prop `isSaving`.

For the sake of illustration, I've created a PR targeting this branch - please take a look and see what you think: https://github.com/google/site-kit-wp/pull/5829

**Review by techanvil — CHANGES_REQUESTED** [review]
Looks good @hussain-t but I think there's room for a bit of simplification here :)

File: assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.js]: Thanks, @techanvil. I have approved and merged your changes. I have added a story to the `SpinnerButton` component to show the spinner.

**Review by techanvil — APPROVED** [review]
LGTM, nice one @hussain-t!

**Pull request merged**